aea/crypto/base.py

```diff
@@ -129,9 +119,70 @@ def sign_transaction(self, transaction: Any) -> Any:
         :return: signed transaction
         """
 
+    @abstractmethod
+    def dump(self, fp: BinaryIO) -> None:
+        """
+        Serialize crypto object as binary stream to `fp` (a `.write()`-supporting file-like object).
+
+        :param fp: the output file pointer. Must be set in binary mode (mode='wb')
+        :return: None
+        """
+
+
+class Helper(ABC):
```

Comment:
We now separate between the non-network related methods and the rest

Reply:
`is_transaction_valid` in theory would require to check whether the nonce is correct, hence the need to access to the network. Without that check, I think it is OK

the same for `generate_tx_nonce`? It needs to access the network like Web3 and `getTransactionCount`

Reply:
`generate_tx_nonce` is entirely unrelated to the ledger nonce. It is just a framework side nonce to keep otherwise identical negotiations apart.

`is_transaction_valid` compares the negotiated terms against what has been already settled on the ledger; so it compares contents of two dicts essentially.